docs/02-app/02-api-reference/05-next-config-js/cssChunking.mdx

```diff
@@ -0,0 +1,38 @@
+---
+title: cssChunking (Experimental)
+description: Use the `cssChunking` option to control how CSS files are chunked in your Next.js application.
+---
+
+CSS Chunking is a strategy used to improve the performance of your web application by splitting and re-ordering CSS files into chunks. This allows you to load only the CSS that is needed for a specific route, instead of loading all the application's CSS at once.
+
+You can control how CSS files are chunked using the `experimental.cssChunking` option in your `next.config.js` file:
+
+```tsx filename="next.config.ts" switcher
+import type { NextConfig } from 'next'
+
+const nextConfig = {
+  experimental: {
+    cssChunking: 'loose', // default
+  },
+} satisfies NextConfig
+
+export default nextConfig
+```
+
+```js filename="next.config.js" switcher
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  experimental: {
+    cssChunking: 'loose', // default
+  },
+}
+
+module.exports = nextConfig
+```
+
+## Options
+
+- **`'loose'` (default)**: Next.js will try to merge CSS files whenever possible, guessing the dependencies between files to reduce the number of chunks and therefore the number of requests.
+- **`'strict'`**: Next.js will merge CSS files in the correct order they are imported into your application, which can lead to more chunks and requests.
+
+You may consider using `'strict'` if you run into CSS issues where the order of CSS files is important. However, we recommend using `'loose'` for most applications, as it leads to fewer requests and better performance.
```

Comment:
I'm not sure we should use `strict` because something with the ordering of the CSS. I think it's more because if the "loose" merging resulting in an edge case that broke somebody's CSS (but that doesn't necessarily mean it broke the order).

Reply:
What are the cases when "loose" might break someone's CSS?

Reply:
> What are the cases when "loose" might break someone's CSS?

When page 1 imports a.css and b.css in that order and page 2 imports b.css and a.css in that order.

Loose mode would merge that together in any order, assuming there is no dependency between these two css files.

Strict mode would not merge them together and loads them in a -> b on page 1 and b -> a on page 2.

Note that this scenario only makes sense if there is no client side navigation possible between page 1 and page 2, since client side navigation won't change existing stylesheet, so it won't reorder the stylesheets when navigating from page 1 to 2.

Reply:
thank you for the context, this is very helpful!